src/tests.rs

```diff
@@ -189,6 +196,27 @@ fn decompress_copy_close_to_end_2() {
     assert_eq!(decompressed, &*depress(&buf));
 }
 
+// The `read::FrameEncoder` code uses different code paths depending on buffer
+// size, so let's test both. Also, very small buffers are a good stress test.
+#[test]
+fn read_frame_encoder_big_and_little_buffers() {
```

Comment:
This test case tries to verify some buffer copying optimizations in the new code, by trying with a huge output buffer and a ridiculously tiny one, and making sure we get the same results